src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncExceptionHandlerRewriter.cs

```diff
@@ -28,6 +28,7 @@ internal sealed class AsyncExceptionHandlerRewriter : BoundTreeRewriterWithStack
         private AwaitCatchFrame _currentAwaitCatchFrame;
         private AwaitFinallyFrame _currentAwaitFinallyFrame = new AwaitFinallyFrame();
         private bool _inCatchWithoutAwaits;
+        private bool _needsFinalThrow;
```

Comment:
Not sure where to leave this comment, but there's a PROTOTYPE comment in `RuntimeAsyncRewriter` that can now be removed: `// PROTOTYPE: try/finally rewriting`
Also, it looks like `RuntimeAsyncRewriter._compilation` is never read (can be removed)